src/init.cpp

```diff
@@ -263,6 +265,9 @@ void Shutdown()
     globalVerifyHandle.reset();
     ECC_Stop();
     LogPrintf("%s: done\n", __func__);
+    if (pTracingHandle) {
+        tracing_free(pTracingHandle);
```

Comment:
Are all other threads guaranteed to be shut down here?

Reply:
Yes, `Shutdown()` is the last function called before process end:
https://github.com/zcash/zcash/blob/d6f527a88f145fa5d233e2053eee5650775600b7/src/bitcoind.cpp#L182-L194

Reply:
The design of shutdown code seems to distribute responsibility for ensuring that threads have exited in a way that is hard to verify, so I am not confident that it is free of race conditions. (In fact we know that there are sometimes segfaults on shutdown.) However, it doesn't seem as though this change is making things any worse, so it's not a blocker.

Reply:
Note also that `ptracingHandle` is only used by `setlogfilter`, and at this point all async RPC workers have stopped (due to the `StopRPC()` call earlier in `Shutdown()`.